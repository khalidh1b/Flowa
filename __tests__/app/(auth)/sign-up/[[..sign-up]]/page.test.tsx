import { render, screen } from "@testing-library/react";
import SignUpPage from "@/app/(auth)/sign-up/[[...sign-up]]/page";
import React from "react";

jest.mock("@clerk/nextjs", () => ({
    SignUp: () => React.createElement("div", null, "mock signup component"),
}));

describe("SignUpPage", () => {
    
    //unit test
    it("renders the signup component", () => {
        render(<SignUpPage/>)
        expect(screen.getByText("mock signup component")).toBeInTheDocument()
    })

    //snapshot test
    it("matches snapshot", () => {
        const { asFragment } = render(<SignUpPage/>)
        expect(asFragment()).toMatchSnapshot()
    })
});