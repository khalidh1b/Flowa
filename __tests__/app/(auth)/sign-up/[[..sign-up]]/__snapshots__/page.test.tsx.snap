// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SignUpPage matches snapshot 1`] = `
<DocumentFragment>
  <div>
    mock signup component
  </div>
</DocumentFragment>
`;
